marlowe-dashboard-client/src/Page/Welcome/State.purs

```diff
@@ -49,13 +52,25 @@ mkInitialState walletLibrary =
   { walletLibrary
   , card: Nothing
   , cardOpen: false
-  , walletNicknameOrIdInput: InputField.mkInitialState Nothing
   , walletNicknameInput: InputField.mkInitialState Nothing
+  , walletMnemonicInput: InputField.mkInitialState Nothing
   , walletId: PlutusAppId UUID.emptyUUID
   , remoteWalletDetails: NotAsked
   , enteringDashboardState: false
   }
 
+words :: String -> Array String
+words = Array.filter (notEq "") <<< split (Pattern " ")
+
+walletMnemonicError :: Maybe String -> String -> Maybe WalletMnemonicError
+walletMnemonicError invalidFromServer phrase =
+  if Just phrase == invalidFromServer then
+    Just InvalidMnemonicFromServer
+  else if 24 /= (Array.length $ words phrase) then
```

Comment:
Maybe `MnemonicAmountOfWords` could be extended with like `MnemonicAmountOfWords { expected :: Int, got :: Int }` info so info for the user can be more meaningful.